pkgs/applications/misc/oil/default.nix

```diff
@@ -0,0 +1,50 @@
+{ stdenv, lib, fetchurl, jq, peco, makeWrapper }:
+
+stdenv.mkDerivation rec {
+  pname = "oil-buku";
+  version = "0.3.2";
+
+  src = fetchurl {
+    url = "https://github.com/AndreiUlmeyda/oil/archive/refs/tags/${version}.tar.gz";
+    sha256 = "1nwas1v0z2sjfab8ada9k9pf4jw4jn79rzmmn4n69xlfzzf4npq6";
+  };
+
+  passthru.updateScript = [ ./update.sh ];
+
+  postUnpack = ''
+    sed -i 's/$LIBDIR/$LOCALLIB/g' oil-${version}/src/oil
+  '';
+
+  buildInputs = [ jq peco ];
+  nativeBuildInputs = [ makeWrapper ];
+  dontPatchShebangs = true;
```

Comment:
You man why dontPatchShebangs? Well it simply throws an error otherwise, dunno how should I phrase it but I will try.

Reply:
> You man why dontPatchShebangs? Well it simply throws an error otherwise, dunno how should I phrase it but I will try.

Yeah, that's what I meant. Seems odd that that'd break the program, would love to see the error

Reply:
Here is the output for the build without the dontPatchShebangs = true;
```
nixpkgs [oil●] % nix-build -A oil-buku
--------------------------------------------------------
these derivations will be built:
  /nix/store/1ywk9rrcff40spxxj0mppj8jm6wm3s4h-oil-buku-0.3.2.drv
building '/nix/store/1ywk9rrcff40spxxj0mppj8jm6wm3s4h-oil-buku-0.3.2.drv'...
unpacking sources
unpacking source archive /nix/store/cf35a2jmlcmyzhb2ikkrldszjldhcz85-source
source root is source
patching sources
configuring
no configure script, doing nothing
installing
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/lf8jkld6szj7afs31dha5ckqzc5qm326-oil-buku-0.3.2
strip is /nix/store/pja9g36cy32z3d51942jqk91a6l2d5nv-gcc-wrapper-10.3.0/bin/strip
stripping (with command strip and flags -S) in /nix/store/lf8jkld6szj7afs31dha5ckqzc5qm326-oil-buku-0.3.2/lib  /nix/store/lf8jkld6szj7afs31dha5ckqzc5qm326-oil-buku-0.3.2/bin
patching script interpreter paths in /nix/store/lf8jkld6szj7afs31dha5ckqzc5qm326-oil-buku-0.3.2
/nix/store/lf8jkld6szj7afs31dha5ckqzc5qm326-oil-buku-0.3.2/bin/.oil-wrapped: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/4nmqxajzaf60yjribkgvj5j54x9yvr1r-bash-5.1-p12/bin/bash"
/nix/store/lf8jkld6szj7afs31dha5ckqzc5qm326-oil-buku-0.3.2/lib/json-to-line.jq: unsupported interpreter directive "#!/usr/bin/env -S jq -fr" (set dontPatchShebangs=1 and handle shebang patching yourself)
builder for '/nix/store/1ywk9rrcff40spxxj0mppj8jm6wm3s4h-oil-buku-0.3.2.drv' failed with exit code 1
error: build of '/nix/store/1ywk9rrcff40spxxj0mppj8jm6wm3s4h-oil-buku-0.3.2.drv' failed
```

Reply:
Oh, now that im looking at this more closely, it seems like we don't need what my comment above does. The error is from a completely unrelated file, which we need to patch instead:
```bash
substituteInPlace src/json-to-line.jq --replace \
  "/usr/bin/env -S jq" "${jq}"
```
The program tries to execute `jq` directly, which the automatic shebang patching mechanism doesn't understand. It does this for awk as well in `src/oil/format-columns.awk`, which you would need to patch the same way. See the [source](https://github.com/AndreiUlmeyda/oil/blob/a4db70e8afe9e3e7218f603f04c77ac3b30fc7a6/src/format-columns.awk#L1).

 With both of those files patched, we shouldn't need `dontPatchShebangs` anymore .This way we can also remove the wrapping of `PATH` to make the dependencies available in the install phase, which is nice :)

Reply:
This worked for jq, but I'm not being capable of doing it for awk, since I dunno how to reference awk from nixpkgs :/

Reply:
It seems to be at  `${gawk}/bin/awk`, I think it standads for gnu awk

Reply:
I managed to figure it out eventually haha, made the push already!